### ISSUE TITLE
Remote Kernel Console has long (several seconds) delay before showing the '(' character

### ISSUE THREAD

**Issue by MikeFair** [spyder-ide/spyder#14893]
### Issue Report Checklist

* [x ] Searched the [issues page](https://github.com/spyder-ide/spyder/issues?q=is%3Aissue) for similar reports
* [ x] Read the relevant sections of the [Spyder Troubleshooting Guide](https://github.com/spyder-ide/spyder/wiki/Troubleshooting-Guide-and-FAQ) and followed its advice
* [ x] Reproduced the issue after updating with ``conda update spyder`` (or ``pip``, if not using Anaconda)
* [ ] Could not reproduce inside ``jupyter qtconsole`` (if console-related)
* [ x] Tried basic troubleshooting (if a bug/error)
    * [x ] Restarted Spyder
    * [ ] Reset preferences with ``spyder --reset``
    * [ ] Reinstalled the latest version of [Anaconda](https://www.anaconda.com/download/)
    * [ x] Tried the other applicable steps from the Troubleshooting Guide
* [ ] Completed the **Problem Description**, **Steps to Reproduce** and **Version** sections below

## Problem Description
I connect to a remote kernel using a local Spyder console via localhost port forwarding on a Putty console connection.

Whenever I type the '(' character where it could be interpreted as a function opener there is a delay of several seconds before the character appears.  This is obviously related to the help tab as once the character appears the "no documentation available" message pops up.  The '(' characters will also "queue up" so when it doesn't appear and you type it again, you will get multiple delayed results.

Turning off the "IPython Console" automatic connections feature in the Settings->Help page stops the problem, but that obviously turns that feature off for _all_ consoles.

The behavior is async as I can still type other characters, it will then put the '(' character wherever the cursor is at the time the timeout expires.

### What steps reproduce the problem?

1. Start a Kernel on the remote system using --ip=127.0.0.1 and snag the json configuration file
2. Use SSH port forwarding (I used the SSH->Connection->Tunnels config) to forward all ports in the config to the localhost ports
3. Copy the JSON config locally
4. "Connect to an existing kernel" in Spyder, use the JSON config (bc of port forwards, no remote host config required)
5. type a few characters then hit '(' 

### What is the expected output? What do you see instead?
Expected: '(' immediately upon typing 
Actual: There is a delay of several seconds, the '(' appears wherever the cursor is at that moment

## Versions

* Spyder version: 4.2.3 (Latest as of submission)
* Python version: 3.8.5
* Qt version: 
* PyQt version: 
* Operating System name/version: 
Windows 10 
Remote IPython Kernel 1.10.2 (A RedHat based system not in my control)

**Comment by steff456** [comment]
Hi @MikeFair,

Can you try to see if there's a difference if your computer is connected to cable vs wifi?

**Comment by ccordoba12** [comment]
Hey @MikeFair, this happens when you have very objects in the kernel because it takes a long time to introspect them. The only known solution is to go to the menu

    Tools > Preferences > IPython console > Display

and deactivate the option called `Show calltips`.

Please do that and let us know if it works for you.

**Comment by MikeFair** [comment]
@ccordoba12 Umm, in my case it was a brand new remote kernel that I had just started up and had only added one string hello_str = "Hello World!".  Later on I had added an np_array of 50 random ints... 

I'll try disabling the "Show calltips" as you mentioned, and as I mentioned, it also goes away if I clear the "IPython Console" optin in "automatic connections" section.

@steff456 I unfortunately don't have easy access to a wired connection, only WiFi through the apartment building I'm staying in.  I'll see what I can do though.
